src/main/java/org/apache/accumulo/access/Authorizations.java

```diff
@@ -33,70 +33,24 @@
  *
  * @since 1.0.0
  */
-public final class Authorizations implements Iterable<String>, Serializable {
+public final record Authorizations(Set<String> authorizations)
+    implements Iterable<String>, Serializable {
 
   private static final long serialVersionUID = 1L;
 
   private static final Authorizations EMPTY = new Authorizations(Set.of());
 
-  private final Set<String> authorizations;
-
-  private Authorizations(Set<String> authorizations) {
+  public Authorizations(Set<String> authorizations) {
     this.authorizations = Set.copyOf(authorizations);
   }
 
-  /**
-   * Returns the set of authorization strings in this Authorization object
-   *
-   * @return immutable set of authorization strings
-   */
-  public Set<String> asSet() {
-    return authorizations;
-  }
-
-  @Override
-  public boolean equals(Object o) {
-    if (o instanceof Authorizations) {
-      var oa = (Authorizations) o;
-      return authorizations.equals(oa.authorizations);
-    }
-
-    return false;
-  }
-
-  @Override
-  public int hashCode() {
-    return authorizations.hashCode();
-  }
-
   @Override
-  public String toString() {
-    return authorizations.toString();
+  public Iterator<String> iterator() {
+    return authorizations.iterator();
   }
 
-  /**
-   * @return a pre-allocated empty Authorizations object
-   */
-  public static Authorizations of() {
+  public static Authorizations empty() {
     return EMPTY;
   }
 
-  /**
-   * Creates an Authorizations object from the set of input authorization strings.
-   *
-   * @param authorizations set of authorization strings
-   * @return Authorizations object
-   */
-  public static Authorizations of(Set<String> authorizations) {
-    if (authorizations.isEmpty()) {
-      return EMPTY;
```

Comment:
Removing this we lose retruning EMPTY when trying to create w/ an empty set.   To get this same functionality the user would always need to do the following.

```java
Set<String> as.;
Authorizations auths; 
if(as.isEmpty()){
    auths = Authorizations.empty();
}else{
  auths = new Authorizations(as);
}
```